### PR TITLE
GRPC dependencies updates

### PR DESCRIPTION
**Pull request by olegbespalov**
# What?

This PR brings updates of the dependencies with the fixes (see commits breakdown):
* the first fix is to frizes the grpc testing proto as the k6 package, which removes the hard coupling to the dependency version and unlocks updating of the dependencies
* second fix is also around the logic in tests that could cause the issue with newer versions of the `github.com/jhump/protoreflect`

# Why?

Closes: #3069

### CODE REVIEW

File: lib/netext/grpcext/conn_test.go [hunk 1]
```diff
@@ -259,9 +259,17 @@ func (m *getServiceFileDescriptorMock) Recv() (*reflectpb.ServerReflectionRespon
...
+	path := "any-path"
 	parser := protoparse.Parser{
 		InferImportPaths: false,
 		Accessor: protoparse.FileAccessor(func(filename string) (io.ReadCloser, error) {
+			// a small hack to make sure we are parsing the right file
+			// otherwise the parser will try to parse "google/protobuf/descriptor.proto"
+			// with exactly the same name as the one we are trying to parse for testing
+			if filename != path {
+				return nil, nil
+			}
```
Comment (mstoykov) [inline lib/netext/grpcext/conn_test.go:271]: Why is it trying to load a different file?

**Review by mstoykov — APPROVED** [review]
LGTM! although I feel like we now have 3 different packages with protosets and am not certain if we should have 3 different ones :shrug:

**Review by imiric — APPROVED** [review]
I didn't go through the vendored changes, but the ones in our repo look good. :+1: Besides that hack that Mihail mentioned, which is not very clear.

Do you know what happened to the `test.proto` file after it was deleted in https://github.com/grpc/grpc-go/pull/6164? Is it now embedded in the generated `pb.go` files? I couldn't figure out from the PR how they're testing with the new package.

It sucks that we now have to maintain `grpc_testing` ourselves, but it's better to have control over it, and we're forced to anyway :-/

File: lib/netext/grpcext/conn_test.go (hunk 1 above)
Comment (olegbespalov) [reply lib/netext/grpcext/conn_test.go:271]: It's coming from the library update https://github.com/bufbuild/protocompile/commit/7c5114ee37b44b274e401c07b53f3fbfadf81dbb

**Comment by olegbespalov** [comment]
>Do you know what happened to the test.proto file after it was deleted in https://github.com/grpc/grpc-go/pull/6164? Is it now embedded in the generated pb.go files? I couldn't figure out from the PR how they're testing with the new package.

Well, yeah, it seems like the source proto file was just deleted, we could maybe check with authors (or maybe just open the PR to them)

https://github.com/grpc/grpc-go/blob/master/interop/grpc_testing/test.pb.go#L22

**Pull request merged**